ulog/log.go

```diff
@@ -99,12 +107,57 @@ func TestingLogger() (Log, *spy.Sink) {
 	}, sink
 }
 
-// Logger returns the package-level logger configured for the service
-// TODO:(at) Remove Logger() call, _std and defaultLogger() access in ulog
-func Logger() Log {
+func logger() Log {
+	_setupMu.RLock()
+	defer _setupMu.RUnlock()
+
 	return &baseLogger{
-		log: _std.(*baseLogger).log,
+		log: _std.log,
+	}
+}
+
+// SetLogger sets configured logger at the start of the service
+func SetLogger(log Log) {
+	_setupMu.Lock()
+	defer _setupMu.Unlock()
+	_std = log.(*baseLogger)
+}
+
+// Logger is the context based logger
+func Logger(ctx context.Context) Log {
+	// TODO(anup): Do we want users to access Logger(nil)?
+	if ctx == nil {
+		return logger()
```

Comment:
Looks like we have an allocation, each time we call Log()..

Could you add a TODO to improve it?